api/v1beta1/vmuser_types.go

```diff
@@ -78,6 +78,9 @@ type TargetRef struct {
 	// https://docs.victoriametrics.com/vmauth.html#ip-filters
 	// +optional
 	IPFilters VMUserIPFilters `json:"ip_filters,omitempty"`
+	// BasicAuth allow an endpoint to authenticate over basic authentication
+	// +optional
+	BasicAuth *BasicAuth `json:"basicAuth,omitempty"`
```

Comment:
I think we don't need to reuse `BasicAuth` here, the only field missing in vmuser now is `Username v1.SecretKeySelector ` and we can add it directly to `VMUserSpec`.
`Password v1.SecretKeySelector` in BasicAuth should already be covered by `PasswordRef *v1.SecretKeySelector` https://github.com/VictoriaMetrics/operator/blob/0628def2e2821a860fc3a9bf4f911f80cabb8cd2/api/v1beta1/vmuser_types.go#L26-L27
And `PasswordFile` field is not used.

Reply:
I think theres some misunderstanding about what this feature is. As far as I have used, vmuser can contains multiple path routes and each of that paths includes some hosts as `backend`(but we use one host on each targetRef entry noramlly), we want to implement backends basic authentication, how can set distinct basic auth on these endpoint with some field that has appeared on top level field like vmuser's spec itself?
I have built a custom operator based on the changes you see and its suits our needs for a long time, but we want to keep it up to date with victoriametrics and use officially released version.

Reply:
Oh, sorry, you're right!
But I still think we can add `Username v1.SecretKeySelector` and `Password v1.SecretKeySelector` directly instead of reuse `BasicAuth` since `PasswordFile` is not used.

Reply:
>It will be a little confusing and not straightforward to know they are for basicAuth? 

I think with description, it's ok. 
I'm more against introducing invalid field, which looks like a bug.
What do you think @f41gh7

Reply:
I agree, it's better to use username and password directly. Or wrap it into new struct `TargetRefAuth`.

 It must have  `username` and `password` for basic Authorization. 

 E.g.

```go
type TargetRefAuth struct{
  Username v1.SecretKeySelector
  Password v1.SecretKeySelector
}
```
 
 I think, it less confusing.

Reply:
I think It's good enough to proceed.
I'll go ahead and update the code with the suggested interface. 
Do you Agree?

Reply:
@mohammadkhavari yes, please go ahead!